src/main/credentialServer/httpCredentialHelperServer.ts

```diff
@@ -139,38 +120,33 @@ export class HttpCredentialHelperServer {
     }
   }
 
-  protected lookupCommand(method: string, commandName: string): dispatchFunctionType | undefined {
-    if (commandName) {
-      return this.dispatchTable[method]?.[commandName];
-    }
-
-    return undefined;
-  }
-
-  async get(helperName: string, data: string, request: http.IncomingMessage, response: http.ServerResponse): Promise<void> {
-    return await this.doNamedCommand(requireJSONOutput, helperName, 'get', data, request, response);
-  }
-
-  async list(helperName: string, data: string, request: http.IncomingMessage, response: http.ServerResponse): Promise<void> {
-    return await this.doNamedCommand(requireJSONOutput, helperName, 'list', data, request, response);
-  }
-
-  async erase(helperName: string, data: string, request: http.IncomingMessage, response: http.ServerResponse): Promise<void> {
-    return await this.doNamedCommand(requireNoOutput, helperName, 'erase', data, request, response);
-  }
-
-  async store(helperName: string, data: string, request: http.IncomingMessage, response: http.ServerResponse): Promise<void> {
-    return await this.doNamedCommand(requireNoOutput, helperName, 'store', data, request, response);
-  }
-
-  protected async doNamedCommand(outputChecker: checkFunctionOutputType,
+  protected async doRequest(
     helperName: string,
     commandName: string,
     data: string,
     request: http.IncomingMessage,
     response: http.ServerResponse): Promise<void> {
     let stderr: string;
-    let error: any;
+    let requestCheckError: any = null;
+    const checkers: Record<string, checkerFnType> = {
+      list:  requireJSONOutput,
+      get:   requireJSONOutput,
+      erase: requireNoOutput,
+      store: requireNoOutput,
+    };
+    const checkerFn: checkerFnType|undefined = checkers[commandName];
+
+    if (request.method !== 'POST') {
+      requestCheckError = `Expecting a POST method for the credential-server list request, received ${ request.method }`;
+    } else if (!checkerFn) {
+      requestCheckError = `Unknown credential action '${ commandName }' for the credential-server, must be one of [${ Object.keys(checkers).join('|') }]`;
```

Comment:
Should we always sort the command names? (Otherwise the order is technically undefined, and you're relying on the order in a test.)